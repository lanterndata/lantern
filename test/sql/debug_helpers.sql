CREATE EXTENSION IF NOT EXISTS vector;
CREATE EXTENSION IF NOT EXISTS lanterndb;

-- initially, before we create an index, the variable does not exist
\ir sql/test_helpers/small_world.sql
SHOW hnsw.init_k;

CREATE INDEX ON small_world USING hnsw (vector);
-- verify that the index was created
SELECT * FROM ldb_get_indexes('small_world');

-- it exists after we create an index
SHOW hnsw.init_k;

SET hnsw.init_k = 45;
SHOW hnsw.init_k;
RESET ALL;
SHOW hnsw.init_k;
