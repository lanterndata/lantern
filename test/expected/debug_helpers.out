-- initially, before we create an index, the variable does not exist
\ir sql/test_helpers/small_world.sql
-- creates a simple vector table
CREATE TABLE
  small_world (id varchar(3), vector vector (3));
INSERT INTO
  small_world (id, vector)
VALUES
  ('000', '[0,0,0]'),
  ('001', '[0,0,1]'),
  ('010', '[0,1,0]'),
  ('011', '[0,1,1]'),
  ('100', '[1,0,0]'),
  ('101', '[1,0,1]'),
  ('110', '[1,1,0]'),
  ('111', '[1,1,1]');
\set ON_ERROR_STOP off
SHOW hnsw.init_k;
ERROR:  unrecognized configuration parameter "hnsw.init_k"
\set ON_ERROR_STOP on
CREATE INDEX ON small_world USING hnsw (vector);
INFO:  done init usearch index
INFO:  inserted 8 elements
INFO:  done saving 8 vectors
-- verify that the index was created
SELECT * FROM ldb_get_indexes('small_world');
       indexname        | size  |                                   indexdef                                    
------------------------+-------+-------------------------------------------------------------------------------
 small_world_vector_idx | 24 kB | CREATE INDEX small_world_vector_idx ON public.small_world USING hnsw (vector)
(1 row)

-- it exists after we create an index
SHOW hnsw.init_k;
 hnsw.init_k 
-------------
 10
(1 row)

SET hnsw.init_k = 45;
SHOW hnsw.init_k;
 hnsw.init_k 
-------------
 45
(1 row)

RESET ALL;
SHOW hnsw.init_k;
 hnsw.init_k 
-------------
 10
(1 row)

